pkg/controller/workload/job/job_controller.go

```diff
@@ -165,6 +230,7 @@ func (r *JobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		log.Error(err, "Getting existing workloads")
```

Comment:
I think we need to adjust the line above that says:
```golang
if queueName(&job) == "" && !r.manageJobsWithoutQueueName
```

We still need to reconcile jobs that have the `parent-workload` annotation.

Reply:
This is a blocker for the PR

Reply:
Done, this was actually a bug, that is not covered with an integration test that failed before the fix.